### PR TITLE
Add dependency tracker for chronon group_bys and joins

### PR DESCRIPTION
**Pull request by yuli-han**
## Summary

Add a function to pull all the downstream for chronon group_bys/joins, which can be used as a reference for group_bys/joins deprecation.

For group_bys: pull all the join names which are using the groupBy as join part
For joins: pull all the group_bys names which are using the join as join source.

## Test Plan

- [x] Added Unit Tests
- [ ] Covered by existing CI
- [ ] Integration tested

## Checklist
- [ ] Documentation update

## Reviewers
@hzding621 @pengyu-hou

### CODE REVIEW

File: api/py/ai/chronon/repo/dependency_tracker.py [hunk 1]
```diff
@@ -0,0 +1,85 @@
...
+        self.logger = get_logger(log_level)
+        self.chronon_root_path = chronon_root_path
+
+    def check_join_downstream(self, conf_path) -> List[str]:
+        joins = extract_json_confs(
+            Join,
+            os.path.join(self.chronon_root_path, conf_path))
+        downstreams = []
+        if len(joins) == 0:
+            print(f"No joins found in {conf_path}")
```
Comment (pengyu-hou) [inline api/py/ai/chronon/repo/dependency_tracker.py:39]: use logger? I see you have logger at line 30?

File: api/py/ai/chronon/repo/dependency_tracker.py [hunk 2]
```diff
@@ -0,0 +1,85 @@
...
+        joins = extract_json_confs(
+            Join,
+            os.path.join(self.chronon_root_path, JOIN_FOLDER_NAME))
+        for join in joins:
+            for join_part in join.joinParts:
+                if join_part.groupBy.metaData.name == group_by_name:
+                    downstreams.append(join.metaData.name)
+        return downstreams
+
+    def check_downstream(self, conf_path: str) -> List[str]:
```
Comment (pengyu-hou) [inline api/py/ai/chronon/repo/dependency_tracker.py:77]: do you need a main method for the script?

**Review by pengyu-hou — APPROVED** [review]
LGTM with some minor comments. 

It'd be nice if we could visualize the downstream usage.

**Review by pengyu-hou — APPROVED** [review]
LGTM with some minor comments. 

It'd be nice if we could visualize the downstream usage.

**Review by pengyu-hou — APPROVED** [review]
LGTM with some minor comments. 

It'd be nice if we could visualize the downstream usage.

**Review by pengyu-hou — APPROVED** [review]
LGTM with some minor comments. 

It'd be nice if we could visualize the downstream usage.

File: api/py/ai/chronon/repo/dependency_tracker.py [hunk 3]
```diff
@@ -0,0 +1,85 @@
...
+    def check_join_downstream(self, conf_path) -> List[str]:
+        joins = extract_json_confs(
+            Join,
+            os.path.join(self.chronon_root_path, conf_path))
+        downstreams = []
+        if len(joins) == 0:
+            print(f"No joins found in {conf_path}")
+            return downstreams
+        elif len(joins) > 1:
+            print(f"Multiple joins found in {conf_path}")
```
Comment (hzding621) [inline api/py/ai/chronon/repo/dependency_tracker.py:42]: should we actually raise exception at line 42 and 39? similarly for line 61, 64?

**Review by hzding621 — APPROVED** [review]
Looks great with minor comments

File: api/py/ai/chronon/repo/dependency_tracker.py (hunk 3 above)
Comment (yuli-han) [reply api/py/ai/chronon/repo/dependency_tracker.py]: updated

**Pull request merged**